General/SortingCollections.md

```diff
@@ -0,0 +1,96 @@
+# Sorting collections
+
+A common requirement in software engineering is to sort collections. This page will give information about collection sorting in Pharo.
+
+- [Sorting API](#sorting-api)
+  * [Sort a collection](#sort-a-collection)
+  * [Keep a collection sorted](#keep-a-collection-sorted)
+- [Sort functions](#sort-functions)
+- [Sorting via a block](#sorting-via-a-block)
+
+## Sorting API
+
+Collections in Pharo comes with an API to sort them. In this section, we will present some of it and explain their difference.
+
+The default sorting implemented on Pharo's collection is a merge sort. Mergesort is a worst-case O(N log N) sorting algorithm that usually does only half as many comparisons as heapsort or quicksort.
+
+### Sort a collection 
+
+The two main methods to sort a collection are `#sort:` and `#sorted:`. Those two methods are taking a block or a sort function as parameter (those are explained later in this page) and return a collection sorted based on the parameter.
+The difference between the two methods is that `#sort:` sort the receiver when the `#sorted:` method sort a copy of the receiver.
+
+```Smalltalk
+#(1 2 4 7 3 6 4) sort: #yourself ascending. "#(1 2 3 4 4 6 7) <= This result is the receiver"
+#(1 2 4 7 3 6 4) sorted: #yourself ascending. "#(1 2 3 4 4 6 7) <= This result is a copy of the receiver"
+```
+
+Those two methods also have an equivalent without argument, `#sort` and `#sorted` that sort the collection using the `#<=` comparison operator.
+
+The API described here will sort a collection at one point but new elements added to the collection will not be sorted. If you wish to keep a collection sorted, you should use `SortedCollection` as explained in the next section.
+
```

Comment:
```suggestion

> Warning: If not absolutely needed, one should not use `SortedCollection` because it might impact performances of your software when adding and removing items to the collection.
```